### PR TITLE
[BUG] Buttons section not visible in a large dialog window

### PR DESCRIPTION
**Pull request by pave1-semenov**
**Describe the bug**
Can't see the footer (buttons) in a dialog when the content section is too large. The content section is scrollable, but the container element isn't

**To Reproduce**
1. Open a repair dialog for an item with a large amount of components
2. You'll see the header and components section, but the footer buttons are not visible

**Expected behavior**
The dialog can be scrolled to reveal the buttons

**Screenshots**
Before the fix:
image

After the fix:
image

**Versions (please complete the following information):**
 - Foundry: v12.331
 - System: v12.15.2

**Additional context**
I'm not sure if introducing a system-wide (non-repair-specific) CSS class is the best approach, but it could be helpful for other dialog use cases as well

### CODE REVIEW

**Comment by Stexinator** [comment]
Can you structure it so that only the component list is scrollable?
I think you should have three general areas:

- Item identifying attributes and DC
- scrollable list of components (bonus points if you make that a partial so it can be used for eg Hexes, Rituals and standard crafting later on)
- the button row

**Comment by pave1-semenov** [comment]
Hi @Stexinator! 
I moved the components list into the separate partial with some adjustments to make it easier to use.
As for the dialog - I made the header section also scrollable, cause it can be quite big for a full-cover armor with all the locations damaged 
image

Tried to make both header and list responsible to the viewport size - the calculation is not precise, but should generally work

File: module/item/mixins/costEditMixin.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export let costEditMixin = {
+    attachHtmlListeners(html, callback, ...args) {
+        $(html).find('.component-cost').on('change', () => {
```
Comment (Stexinator) [inline module/item/mixins/costEditMixin.js:3]: There are rumors that jQuery will not be bundled with Foundry after v16 anymore. Do you want to tackle it already?

**Review by Stexinator — APPROVED** [review]

**Comment by Stexinator** [comment]
I will merge it end of the week if you dont want to add additional changes

File: module/item/mixins/costEditMixin.js (hunk 1 above)
Comment (pave1-semenov) [reply module/item/mixins/costEditMixin.js]: Done 👍

**Pull request merged**